core/src/test/java/org/infinispan/eviction/impl/protostream/PrimitiveArrayCtx.java

```diff
@@ -0,0 +1,222 @@
+package org.infinispan.eviction.impl.protostream;
+
+import static org.infinispan.protostream.FileDescriptorSource.fromString;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.infinispan.protostream.MessageMarshaller;
+import org.infinispan.protostream.SerializationContext;
+import org.infinispan.protostream.SerializationContextInitializer;
+
+/**
+ * Adds support for primitive and primitive wrapper arrays.
+ */
+public class PrimitiveArrayCtx implements SerializationContextInitializer {
```

Comment:
This made me think of ProtoStream support and then I saw that you created https://issues.redhat.com/browse/IPROTO-134 :slightly_smiling_face:

Reply:
Yes, I added it a workaround in the tests only, to force Protostream to support  :smile:

Reply:
Couldn't we just change the test so it doesn't use primitive arrays? :)

Most cache operations won't work with primitive array keys, because the `equals()` implementation is wrong, and many cache operations also require `equals()` to work for values, so I would rather prohibit primitive array keys and values instead of adding special support in protostream.

Reply:
Woudn't those primitive arrays be wrapped in ```WrappedByteArray```?

Reply:
Yes, but that would mean primitive arrays would work or would not work depending on the cache encoding, and I would rather they work or not work the same way in any configuration.

Reply:
Right, I see your point. But this was a very old test and I was just making it pass by offering an alternative that works on protostream. Protostream as a marshaller should support it IMO since every other marshaller in the works does :)

On the Infinispan side, we could document the limitation or even throw an exception if someone tries to store those arrays? In either case, I wouldn't want the change to be hidden inside this PR instead of a separate JIRA to get more visibility, as this is a breaking change

Reply:
I agree, I over-reacted a bit to the IPROTO-134 comment because I don't want to add a new feature just to keep an old test working.

Personally I'd like to throw an exception, but we can also explore non-byte array wrappers.